src/main/java/uk/gov/hmcts/reform/iahearingsapi/domain/mappers/LanguageAndAdjustmentsMapper.java

```diff
@@ -0,0 +1,171 @@
+     * "otherReasonableAdjustmentsDetails" value, any manually typed one (without the "code") will add to
+     * "otherReasonableAdjustmentsDetails" too.
+     * Reasonable Adjustment flags (with flagCode "RA****" or "SM****") except Sign Language Interpreter: the codes
+     * of each will populate the "reasonableAdjustments" value, the comment of each, if present, will add to the
+     * "otherReasonableAdjustmentsDetails" value.
+     * @param asylumCase The asylum case queried for a hearing.
+     * @return           A map with the three values to be written in the response to HMC.
+     */
+    public Map<String, List<String>> getLanguageAndAdjustmentsFields(AsylumCase asylumCase) {
+        Map<String, List<String>> fields = new HashMap<>();
+
+        List<CaseFlagDetail> caseFlags = getAppellantAndWitnessCaseFlags(asylumCase)
+            .stream()
+            .map(StrategicCaseFlag::getDetails)
+            .flatMap(Collection::stream)
+            .filter(detail -> Objects.equals("Active", detail.getCaseFlagValue().getStatus()))
+            .toList();
+
+        List<CaseFlagDetail> languageFlags = new ArrayList<>();
+        List<CaseFlagDetail> reasonableAdjustmentsFlags = new ArrayList<>();
+
+        caseFlags
+            .forEach(flagDetail -> {
+                if (isLanguageCaseFlag(flagDetail)) {
+                    languageFlags.add(flagDetail);
+                } else if (isReasonableAdjustmentFlag(flagDetail)) {
+                    reasonableAdjustmentsFlags.add(flagDetail);
+                }
+            });
+
+        // get the languageCaseFlags sorted to have the ones with code above the ones without
+        List<CaseFlagDetail> sortedLanguageFlags = languageFlags.stream()
+            .sorted(Comparator.comparing(
+                detail -> detail.getCaseFlagValue().getSubTypeKey(),
+                Comparator.nullsLast(Comparator.naturalOrder())))
+            .toList();
+
+        // extract one flag with code to populate "interpreterLanguage" and group all the others with code
+        // and the ones without code to populate "otherReasonableAdjustmentsDetails"
+        CaseFlagDetail interpreterLanguageFlag = null;
+        List<CaseFlagDetail> otherLanguageFlags = new ArrayList<>(Collections.emptyList());
+
+        int i = 0;
+        while (i < sortedLanguageFlags.size()) {
+            CaseFlagDetail flag = sortedLanguageFlags.get(i);
+
+            if (interpreterLanguageFlag == null && flag.getCaseFlagValue().getSubTypeKey() != null) {
+                interpreterLanguageFlag = flag;
+            } else {
+                otherLanguageFlags.add(flag);
+            }
+            i++;
+        }
+
+        String interpreterLanguage = interpreterLanguageFlag != null
+            ? interpreterLanguageFlag.getCaseFlagValue().getSubTypeKey()
+            : "";
+
+        fields.put(INTERPRETER_LANGUAGE, Collections.singletonList(interpreterLanguage));
```

Comment:
Small one but might be clearer to group all three fields that we are putting in "fields" just so others can see all the fields that are being put in at glance. (can always extract the logics to private methods)

Reply:
I moved these three methods to be next to each other at the bottom